### PR TITLE
added install target, $(CXX)

### PR DESCRIPTION
**Pull request by smoe**
I just had a look around - please take this PR just as an anchor to discuss what should happen in general with "make install" when components are configured that are more assigned to the server side of boinc - like these scientific/sample apps.

### CODE REVIEW

File: samples/sleeper/Makefile [hunk 1]
```diff
@@ -16,6 +17,9 @@ PROGS = sleeper
 
 all: $(PROGS)
 
+install: all
+	@echo "--- The example app $(PROGS) is not installed ---"
```
Comment (AenBleidd) [inline samples/sleeper/Makefile:21]: Such kind of message will mislead the user, (s)he will decide that smth wrong happened during installation and it should be fixed somehow.

**Comment by AenBleidd** [comment]
According to your comment, converting this PR to draft.

**Comment by smoe** [comment]
I have removed the message. Since regular users don't see this, I don't ultimately share your concern. Important to me is only that it compiles - which it does not without the install target.

**Comment by smoe** [comment]
I once went through this for the BOINC server package on Debian, or the [boinc-app-examples](https://packages.debian.org/experimental/boinc-app-examples). I suggest to install to /usr/lib/boinc/apps. Maybe you bring this up at the next boinc contributors meeting.

**Comment by AenBleidd** [comment]
Looks fine, thanks. Merging

**Pull request merged**